compiler/z/codegen/OMRTreeEvaluator.cpp

```diff
@@ -14022,26 +14026,32 @@ TR::Register *OMR::Z::TreeEvaluator::inlineVectorUnaryOp(TR::Node *node, TR::Cod
 TR::Register *OMR::Z::TreeEvaluator::inlineVectorBinaryOp(TR::Node *node, TR::CodeGenerator *cg,
     TR::InstOpCode::Mnemonic op)
 {
-    TR_ASSERT(node->getNumChildren() <= 2, "Binary Node must only contain 2 or less children");
+    const bool isMasked = node->getOpCode().isVectorMasked();
+    TR_ASSERT(node->getNumChildren() <= (isMasked ? 3 : 2),
+        "Binary Node must contain no more than 2 children, or up to 3 if a mask is present.");
```

Comment:
Looking at the code below, it looks at least two children are required for an operation without a mask and at least three are required for an operation with a mask.  I think this assertion should change to test for equality if it's needed.